crates/transaction-fuzzer/src/transaction_data_gen.rs

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use move_core_types::account_address::AccountAddress;
+use proptest::arbitrary::*;
+use proptest::prelude::*;
+
+use crate::type_arg_fuzzer::{gen_type_tag, pt_for_tags};
+use proptest::collection::vec;
+use sui_types::base_types::{ObjectID, ObjectRef, SequenceNumber, SuiAddress};
+
+use sui_types::digests::ObjectDigest;
+use sui_types::messages::{
+    GasData, TransactionData, TransactionDataV1, TransactionExpiration, TransactionKind,
+};
+
+use crate::account_universe::{gas_budget_selection_strategy, gas_price_selection_strategy};
+
+const MAX_NUM_GAS_OBJS: usize = 1024_usize;
+
+pub fn gen_transaction_expiration_with_bound(
+    max_epoch: u64,
+) -> impl Strategy<Value = TransactionExpiration> {
+    prop_oneof![
+        Just(TransactionExpiration::None),
+        (0u64..=max_epoch).prop_map(TransactionExpiration::Epoch),
+    ]
+}
+
+pub fn gen_transaction_expiration() -> impl Strategy<Value = TransactionExpiration> {
```

Comment:
my take is that we are better off with a single strategy with weight that with multiple ones. However others (including you) may differ so it's all good ;)